tutorials/dask.md

```diff
@@ -0,0 +1,149 @@
+
+> **Note**: This tutorial is not meant to promote a folder of TIFFs as a "good way" to store large datasets on disk; but it is an undoubtedly common scenario in microscopy.  Chunked formats such as `hdf5` or `zarr` are superior in many ways, but they do require the user to either duplicate their data, or go "all in" and delete the original data after conversion.  And while `napari` can easily handle something like a `zarr` store, it can be a bit more limiting inasmuch as it requires programs that capable of viewing it (i.e. you can't currently simply drag it into Fiji ...)
+
+The first part of this tutorial demonstrates how to use [`dask`](https://docs.dask.org/en/latest/) and [`dask.delayed`](https://docs.dask.org/en/latest/delayed.html) (or [`dask_image`](https://github.com/dask/dask-image)) to feed napari image data "[lazily](https://en.wikipedia.org/wiki/Lazy_evaluation)": that is, the specific image file corresponding to the requested timepoint/channel is only read from disk at the moment it is required (based on the current position of the dimension sliders in `napari`).  Additionally, we will see that *any* function that takes a filepath and returns a `numpy` array can be used to lazily read image data.  This can be useful if you have a proprietary format that is not immediately recognized by napari (but for which you have at least some way of reading into a `numpy` array)
+
+In some cases, data must be further processed prior to viewing, such as a deskewing step in stage-scanning light sheet microscopes.  Or perhaps you'd like to apply some basic image corrections or ratiometry prior to viewing.  The second part of this tutorial demonstrates the use of the [`dask.array.map_blocks`](https://docs.dask.org/en/latest/array-api.html#dask.array.map_blocks) function to describe an arbitrary sequence of functions in a declarative manner that will be performed *on demand* as you explore the data (i.e. move the sliders) in `napari`.
+
+## using dask.delayed to load images
+
+If you have a function that can take a filename and return a `numpy` array, for instance `skimage.io.imread`, you can create a "lazy" version of that function by calling `dask.delayed` on the function itself.  This *new* function, when handed a filename, will not actually read the file until explicitly asked with the `compute()` method:
+
+```python
+from skimage.io import imread
+from dask import delayed
+
+lazy_imread = delayed(imread)
+reader = lazy_imread('/path/to/file.tif')  # doesn't actually read the file
+array = reader.compute()  # *now* it reads.
+```
+
+We can create a [Dask array](https://docs.dask.org/en/latest/array.html) of delayed file-readers for *all* of the files in our multidimensional experiment using the `dask.array.from_delayed` function and a [`glob`](https://docs.python.org/3/library/glob.html) filename pattern:
+
+```python
```

Comment:
Do you know what it would take to get the example to render more "jupyter notebook" style including the cool html pictures of the dask array as in the dask blogs - https://blog.dask.org/2019/08/09/image-itk

Reply:
![image](https://user-images.githubusercontent.com/6531703/69299771-3ba01b00-0bc6-11ea-9c1a-056c7d8d468f.png)

Reply:
Not sure... they may actually be using jupyter somehow for rendering their posts.  Haven’t looked into how you guys are formatting these markdown pages yet, but it would be an easy enough workaround to just put some images in there as a quick and dirty solution

Reply:
Ok yeah, maybe just one or two screenshots with the nice array representation works for now, we can revisit in the future

Reply:
> Do you know what it would take to get the example to render more "jupyter notebook" style including the cool html pictures of the dask array as in the dask blogs - https://blog.dask.org/2019/08/09/image-itk

also... maybe this: https://github.com/jupyter/nbconvert
```jupyter nbconvert --to markdown <input notebook>```
... and then move around some image files